src/test/ui/consts/miri_unleashed/use-static-via-const.rs

```diff
@@ -0,0 +1,29 @@
+// run-pass
+// compile-flags: -Zunleash-the-miri-inside-of-you
+
+static DATA: [u8; 3] = *b"abc";
+
+const REF: &'static [u8] = &DATA;
+//~^ WARN skipping const checks
+
+const PTR: *const u8 = DATA.as_ptr();
+//~^ WARN skipping const checks
+
+fn parse_slice(r: &[u8]) -> bool {
+    match r {
+        REF => true,
+        _ => false,
+    }
+}
+
+fn parse_id(p: *const u8) -> bool {
+    match p {
+        p if p == PTR => true,
+        _ => false,
+    }
+}
+
+fn main() {
+    assert!(parse_slice(&DATA));
+    assert!(parse_id(DATA.as_ptr()));
+}
```

Comment:
This doesn't actually *read* from the static in a const, does it?

Also I feel we are getting a lot of duplicate tests here, some of the ones above seem to already test the same thing?

Reply:
Hm, I added this test because it caught some ICEs that the others didn't- one related to the borrowck change and another related to the fallback-to-ByRef changes. Perhaps that code would also be exercised by some of your suggestions above? I'll look into it.

Reply:
If this caused extra ICEs, that seems fair then. Please add a comment in the test explaining that.